src/main/java/uk/gov/companieshouse/company/metrics/exception/ControllerExceptionHandler.java

```diff
@@ -46,9 +57,43 @@ public ResponseEntity<Object> handleException(Exception ex, WebRequest request)
         responseBody.put("correlationId", correlationId);
         request.setAttribute("javax.servlet.error.exception", ex, 0);
         logger.error("correlationId = " + correlationId, ex);
+
         return new ResponseEntity<>(responseBody, HttpStatus.INTERNAL_SERVER_ERROR);
     }
 
+    @ExceptionHandler(value = {DataAccessResourceFailureException.class})
```

Comment:
[non-blocking] Elsewhere we've gone for the whole parent class of exceptions `DataAccessException`, though it is true that manually brining down the DB after starting the app produces the one in your handler. Would this be sufficient however to catch all possible cases when we'd like to throw service unavailable?

Reply:
Updated handler, unit test still passes.